### PR TITLE
Update AppConfig typespec to match service behavior by returning RequestIdResponseHeader instead of ClientRequestIdHeader.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Testing the service behavior shows that the response headers that come back include `x-ms-request-id` and not `x-ms-client-request-id`, when making a request to the service's `getKeyValue` operation, using the current C++ generated SDK:
```C++
Azure::Response<GetKeyValueResult> response
    = configurationClient.GetKeyValue("key", "accept");
```
response headers from the service

```cmd
> curl "https://<config store name>.azconfig.io/kv/key?api-version=2023-11-01" -H "Content-Type: application/json" -H "Authorization: Bearer <access token> " -i
```
```text
HTTP/1.1 200 OK
Date: Wed, 04 Dec 2024 22:16:58 GMT
Content-Type: application/vnd.microsoft.appconfig.kv+json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
ETag: "qZ..."
Last-Modified: Tue, 03 Dec 2024 00:17:15 GMT
Sync-Token: z...
x-ms-request-id: 15...
x-ms-correlation-request-id: 15...
Strict-Transport-Security: max-age=31536000; includeSubDomains
Access-Control-Allow-Origin: *

... response body
```

Here's the REST API docs around common response headers for AppConfig (thanks @avanigupta for sharing):
https://learn.microsoft.com/en-us/azure/azure-app-configuration/rest-api-headers#response-headers

The issue exists in the swagger version of the specification as well:
https://github.com/Azure/azure-rest-api-specs/blob/f3245a566562790a4a0229cb53761fd1c4125e33/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json#L734-L738

cc @catalinaperalta, @lmazuel, @markcowl, @YalinLi0312, @samsadsam, @jsquire 

JFYI @RickWinter, @antkmsft

### CODE REVIEW

File: specification/appconfiguration/AppConfiguration/routes.tsp
Comment (ahsonkhan) [inline specification/appconfiguration/AppConfiguration/routes.tsp:302]: I am not sure whether this request header should also change, and if so, should we set this to `RequestIdResponseHeader`, or just remove it all together?

The AppConfig Rest API docs don't specify.

**Comment by ahsonkhan** [comment]
Though the `Swagger BreakingChange` CI check is highlighting a breaking change: [1014 - RemovingHeader](https://github.com/Azure/openapi-diff/tree/master/docs/rules/1014.md), from my testing of the actual service, the swagger incorrectly specified which header is being returned from the service.

I'd defer to the expertise of the AppConfig service team here on how we'd like to proceed.

**Comment by avanigupta** [comment]
> Question to the service owners, should we update the expected request header as well below? Or only update the response header?
> 
> https://github.com/Azure/azure-rest-api-specs/blob/c2d07b8aa83bbd091ec6391b8559f2f2a56ef846/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json#L3172-L3181
> 
> cc @avanigupta, @jimmyca15


We should only update the response header.

File: specification/appconfiguration/AppConfiguration/routes.tsp
Comment (avanigupta) [reply specification/appconfiguration/AppConfiguration/routes.tsp:302]: We have it documented here: https://learn.microsoft.com/en-us/azure/azure-app-configuration/rest-api-headers

File: specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json
Comment (avanigupta) [inline specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json:737]:
```suggestion
                "description": "An opaque, globally-unique, server-generated string identifier for the request."
```

File: specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json
Comment (avanigupta) [inline specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json:2263]:
```suggestion
                "description": "An opaque, globally-unique, server-generated string identifier for the request."
```

**Review by avanigupta — APPROVED** [review]

**Review by catalinaperalta — COMMENTED** [review]
The update looks good to me, but we need typespec validation to pass before merging. Looks like you need to regenerate the swagger. If the docs dont match what you're expecting then please open an issue in the typespec-azure repo for the tsp team to update them. We shouldnt be adding any manual changes in the generated swagger.

File: specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json
Comment (ahsonkhan) [inline specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json:2263]: The typespec validation CI step is failing for this change, when compared to what we have in TypeSpec for the `GetSnapshot` operation.

https://github.com/Azure/azure-rest-api-specs/actions/runs/12169538899/job/33942675211?pr=31745

The TypeSpec doesn't specify this header for some reason, so we need to decide whether the right behavior is to add it to TypeSpec to match the behavior. I am not seeing where TypeSpec is specifying the `x-ms-client-request-id` for this operation. I see this being used, but I thought it was ap
https://github.com/Azure/azure-rest-api-specs/blob/00d4638b98a5c4d0c5783b7740e54b9d61784e77/specification/appconfiguration/AppConfiguration/routes.tsp#L501-L534

I am seeing this the `getSnapshot` operator inheriting from `standardOps` which has SupportsClientRequestId, which I am presuming is adding this header to both the request and response:
https://github.com/Azure/azure-rest-api-specs/blob/00d4638b98a5c4d0c5783b7740e54b9d61784e77/specification/appconfiguration/AppConfiguration/routes.tsp#L24-L27

https://azure.github.io/typespec-azure/docs/libraries/azure-core/reference/data-types/#Azure.Core.Traits.SupportsClientRequestId

**For now, I'll pull this change out into a separate issue to resolve and we can come back to it.**

Comment (ahsonkhan) [reply specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json]: https://github.com/Azure/azure-rest-api-specs/issues/31766

**Comment by ahsonkhan** [comment]
> The update looks good to me, but we need typespec validation to pass before merging.

The typespec validation is passing now (I scoped the fix to just the one endpoint for now)

> We shouldnt be adding any manual changes in the generated swagger.

Is the checked-in swagger that I am modifying `stable/2023-11-01/appconfiguration.json` generated from somewhere? Is that the swagger you are suggesting I don't make manual changes to?

File: specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json
Comment (catalinaperalta) [reply specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json]: @ahsonkhan are you saying that the `SupportsClientRequestId` trait is adding x-ms-client-request-id in both the request and response? If so, then we probably need to tweak the template a bit so it's only x-ms-client-request-id on the request and x-ms-request-id on the response.

Comment (catalinaperalta) [reply specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json]: I'd like to understand the issue before suggesting an update to the template

**Comment by catalinaperalta** [comment]
> > The update looks good to me, but we need typespec validation to pass before merging.
> 
> The typespec validation is passing now (I scoped the fix to just the one endpoint for now)

👍

> > Looks like you need to regenerate the swagger.
> > We shouldnt be adding any manual changes in the generated swagger.
> 
> Is the checked-in swagger that I am modifying `stable/2023-11-01/appconfiguration.json` generated from somewhere? Is that the swagger you are suggesting I don't make manual changes to and generate instead?

Yes that's auto-generated when you run `tsp compile .`

**Comment by ahsonkhan** [comment]
After confirming with the service team that this aligns with the expected actual service behavior, and as per @JeffreyRichter's recommendation, applying the "BreakingChange-Approved-Bugfix" label for this change.

**Comment by ahsonkhan** [comment]
@m-nash I am seeing a `typespec-charp` error in the CI step for `SDK azure-sdk-for-net-track2`. I will reach out offline to get some help on how we unblock this.
```text
cmdout	[.Net] Start to call tsp-client to generate package:Azure.Data.AppConfiguration
cmderr	[.Net] ExternalError: Emitter "@azure-tools/typespec-csharp" crashed! This is a bug.
cmderr	[.Net] Write-Error: [ERROR] Failed to generate typespec project:/mnt/vss/_work/1/s/azure-rest-api-specs/specification/appconfiguration/AppConfiguration. Exit code: 1.
cmderr	[.Net] Write-Error: [ERROR] Please review the detail errors for potential fixes.
```

**Comment by m-nash** [comment]
The csharp generation failure is safe to ignore.  The CI is trying to generate with tsp and the project in azure-sdk-for-net is still configured to generate from [swagger](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/autorest.md).  Any issues with tsp generation will be address when we convert this project over in the sdk repo.

File: specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json
Comment (ahsonkhan) [reply specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/2023-11-01/appconfiguration.json:2263]: We decided to fix this in the same PR. Thanks @catalinaperalta for your help.

**Review by catalinaperalta — APPROVED** [review]

**Review by RickWinter — APPROVED** [review]

**Pull request merged**
